cmd/kubeadm/app/util/config/masterconfig.go

```diff
@@ -73,3 +67,19 @@ func setInitDynamicDefaults(cfg *kubeadmapi.MasterConfiguration) error {
 
 	return nil
 }
+
+// TryLoadMasterConfiguration tries to loads a Master configuration from the given file (if defined)
+func TryLoadMasterConfiguration(cfgPath string, cfg *kubeadmapi.MasterConfiguration) error {
```

Comment:
Could you add an unit test for this as well?
1) empty file ``, fail
2) a configmap or something, fail
3) non-existent path, fail
4) bare config
```yaml
kind: MasterConfiguration
apiVersion: kubeadm.k8s.io/v1alpha1
```
should pass
5) config with an option
```yaml
kind: MasterConfiguration
apiVersion: kubeadm.k8s.io/v1alpha1
kubernetesVersion: "latest"
```
should pass

However, consider this optional, since it didn't exist before